web/src/main/angular/src/app/core/components/server-and-agent-list/server-and-agent-list-container.component.ts

```diff
@@ -75,7 +75,9 @@ export class ServerAndAgentListContainerComponent implements OnInit, OnDestroy {
                     return this.serverAndAgentListDataService.getData(appName, range).pipe(
                         filter((res: {[key: string]: IServerAndAgentData[]}) => {
                             if (this.agentId) {
-                                const filteredList = this.filterServerList(res, this.agentId, ({ agentId }: IServerAndAgentData) => this.agentId.toLowerCase() === agentId.toLowerCase());
+                                const filteredList = this.filterServerList(res, this.agentId, ({ agentId, agentName }: IServerAndAgentData) =>
+                                    this.agentId.toLowerCase() === agentId.toLowerCase() || ( agentName && this.agentId.toLowerCase() === agentName.toLowerCase() )
+                                );
```

Comment:
`this.agentId` is derived from url and now its value is only matched to `agentId` so I guess it's no need to compare it with agentName here.